x-pack/plugins/transform/public/app/common/navigation.tsx

```diff
@@ -12,21 +12,31 @@ import rison from 'rison-node';
 import { SECTION_SLUG } from '../constants';
 
 /**
- * Gets a url for navigating to Discover page.
+ * Gets url state for navigating to Discover page with optional link to a specific index pattern.
  * @param indexPatternId Index pattern ID.
- * @param baseUrl Base url.
  */
-export function getDiscoverUrl(indexPatternId: string, baseUrl: string): string {
+export function getDiscoverUrlState(indexPatternId?: string): string {
   const _g = rison.encode({});
 
   // Add the index pattern ID to the appState part of the URL.
-  const _a = rison.encode({
-    index: indexPatternId,
-  });
-
-  const hash = `/discover#?_g=${_g}&_a=${_a}`;
+  const _a = rison.encode(
+    typeof indexPatternId === 'string'
+      ? {
+          index: indexPatternId,
+        }
+      : {}
+  );
+
+  return `_g=${_g}&_a=${_a}`;
+}
 
-  return `${baseUrl}/app${hash}`;
+/**
+ * Gets a url for navigating to Discover page.
+ * @param indexPatternId Index pattern ID.
+ * @param baseUrl Base url.
+ */
+export function getDiscoverUrl(indexPatternId: string, baseUrl: string): string {
+  return `${baseUrl}/app/discover#?${getDiscoverUrlState(indexPatternId)}`;
 }
```

Comment:
discover plugin provides a URL generator https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/discover/public/url_generator.ts#L91

Reply:
Good find, we hadn't used those generators yet in the transform plugin. I refactore to make use of it which, the results is that these utils are not necessary anymore 🥳 - related update in bb30f3f.